File: mergeSort.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
    nums1 = nums1.slice(0, m);
    nums2 = nums2.slice(0, n);
    let i = 0;
    let j = 0;
    while (j < n) {
        const num1 = nums1[i];
        const num2 = nums2[j];
        if (num1 < num2)
            i++;
        else {
            nums1.splice(i++, 0, nums2[j++]);
        }
    }
    console.log(nums1);
};

console.log(
    merge(
        [1, 2, 3, 0, 0, 0],
        3,
        [2, 5, 6],
        3)
);

File: mergeSort.py

def merge(nums1, m, nums2, n):
    """
    :type nums1: List[int]
    :type m: int
    :type nums2: List[int]
    :type n: int
    :rtype: void Do not return anything, modify nums1 in-place instead.
     """
    i = 0
    j = 0
    while j < n:
        print j
        if nums2[j] < nums1[i]:
            nums1.insert(i, nums2[j])
            i += 2
            j += 1
        else:
            i += 1
    print nums1

merge(
    [1, 2, 3, 0, 0, 0],
    3,
    [2, 5, 6],
    3)
